deepspeed/runtime/zero/partition_parameters.py

```diff
@@ -885,6 +894,10 @@ def _allgather_params(self, param_list, hierarchy=0):
         return None
 
     def _reduce_scatter_gradients(self, param_list):
+        rank = self.rank
```

Comment:
This one is fine for now but will probably need to be changed and placed inside the loop on 905, since this is going over a list of params

Reply:
Changed and fixed in the new commit (1f4a749)